pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -1382,3 +1383,54 @@ func (c *K8sOrchestrator) AnnotateVolumeSnapshot(ctx context.Context, volumeSnap
 	volumeSnapshotNamespace string, annotations map[string]string) (bool, error) {
 	return c.updateVolumeSnapshotAnnotations(ctx, volumeSnapshotName, volumeSnapshotNamespace, annotations)
 }
+
+// ConfigMapAlreadyExists checks if ConfigMap with given name already exists in the given namespace.
+// If it exists, this function returns true, otherwise returns false.
+func (c *K8sOrchestrator) ConfigMapAlreadyExists(ctx context.Context, name string, namespace string) bool {
+	log := logger.GetLogger(ctx)
+
+	if _, err := c.k8sClient.CoreV1().ConfigMaps(namespace).Get(ctx, name, metav1.GetOptions{}); err == nil {
+		log.Infof("ConfigMap with name %s already exists in namespace %s", name, namespace)
+		return true
+	}
+	return false
+}
+
+// CreateConfigMap creates the ConfigMap with given name, namespace, data and
+// immutable parameter values if it doesn't exist. If ConfigMap already exists,
+// then it returns the data from existing ConfigMap.
+func (c *K8sOrchestrator) CreateConfigMap(ctx context.Context, name string, namespace string,
```

Comment:
Can we break it down into GetConfigMap and CreateConfigMap?

Reply:
Done